tableauserverclient/models/interval_item.py

```diff
@@ -0,0 +1,175 @@
+        Friday = "Friday"
+        Saturday = "Saturday"
+        LastDay = "LastDay"
+
+
+class HourlyInterval(object):
+    def __init__(self, start_time, end_time, interval_occurrence, interval_value):
+
+        self.start_time = start_time
+        self.end_time = end_time
+        self.frequency = IntervalItem.Frequency.Hourly
+        self.interval = interval_occurrence.lower(), str(interval_value)
+
+    @property
+    def start_time(self):
+        return self._start_time
+
+    @start_time.setter
+    def start_time(self, value):
+        _validate_time(value)
+
+        self._start_time = value
+
+    @property
+    def end_time(self):
+        return self._end_time
+
+    @end_time.setter
+    def end_time(self, value):
+        _validate_time(value)
+        self._end_time = value
+
+    @property
+    def interval(self):
+        return self._interval
+
+    @interval.setter
+    def interval(self, intervals):
+        interval_occurrence, interval_value = intervals
+
+        Hours = IntervalItem.Occurrence.Hours
+        Minutes = IntervalItem.Occurrence.Minutes
+
+        VALID_HOUR_INTERVALS = {1, 2, 4, 6, 8, 12}
+        VALID_MIN_INTERVALS = {15, 30}
+
+        if interval_occurrence not in {Hours, Minutes}:
+            error = "Invalid interval type defined: {}.".format(interval_occurrence)
+            raise ValueError(error)
+        elif interval_occurrence == Hours and int(interval_value) not in VALID_HOUR_INTERVALS:
+            error = "Invalid hour value defined: {}.".format(interval_value)
+            raise ValueError(error)
+        elif interval_occurrence == Minutes and int(interval_value) not in VALID_MIN_INTERVALS:
+            error = "Invalid minute value defined: {}".format(interval_value)
+            raise ValueError(error)
+
+        self._interval = [(interval_occurrence.lower(), str(interval_value))]
+
+
+class DailyInterval(object):
+    def __init__(self, start_time):
+        self.start_time = start_time
+        self.frequency = IntervalItem.Frequency.Daily
+        self.end_time = None
+        self.interval = None
+
+    @property
+    def start_time(self):
+        return self._start_time
+
+    @start_time.setter
+    def start_time(self, value):
+        _validate_time(value)
+        self._start_time = value
+
+
+class WeeklyInterval(object):
+    def __init__(self, start_time, *interval_values):
+        self.start_time = start_time
+        self.frequency = IntervalItem.Frequency.Weekly
+        self.interval = interval_values
+        self.end_time = None
+
+    @property
+    def start_time(self):
+        return self._start_time
+
+    @start_time.setter
+    def start_time(self, value):
+        _validate_time(value)
+        self._start_time = value
+
+    @property
+    def interval(self):
+        return self._interval
+
+    @interval.setter
+    def interval(self, interval_values):
+        if not all(hasattr(IntervalItem.Day, day) for day in interval_values):
```

Comment:
I am not totally gungho on decorators but if we see this pattern again of a list of enums instead of just an enum, I would move this to a decorator to be consistent